torch/csrc/api/src/optim/optimizer.cpp

```diff
@@ -121,12 +121,15 @@ void Optimizer::add_parameters(const std::vector<Tensor>& parameters) {
   parameters_.insert(parameters_.end(), parameters.begin(), parameters.end());
 }
 
-void Optimizer::zero_grad() {
+void Optimizer::zero_grad(bool set_to_none) {
   for (auto& group : param_groups_) {
     for (auto& p : group.params()) {
-      if (p.grad().defined()) {
-        p.grad().detach_();
-        p.grad().zero_();
+      if (p.mutable_grad().defined()) {
```

Comment:
why `mutable_grad` and not `grad` here?

Reply:
Because grad() comes up with an error when using reset that it is const.

The module.cpp zero_grad equivalent also uses mutable_grad().